src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private final int initSize = 10;
+    private T[] strArray;
+    private int pointer = 0;
+
+    public ArrayList() {
+        strArray = (T[]) new Object[initSize];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        strArray[pointer++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureCapacity();
+        pointer++;
+        wrongIndexCheck(index);
+        System.arraycopy(strArray, index, strArray, index + 1, strArray.length - 1 - index);
```

Comment:
```suggestion
        System.arraycopy(strArray, index, strArray, index + 1,  ??? );
```
let's review this part